#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_pic2mpg-debian.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: If mpeg2enc is not available, use ffmpeg to create an MPEG1
## DP: instead. (Debian does not have any MPEG2 encoders).

@DPATCH@
diff -urNad vdr-1.5.16~/PLUGINS/src/pictures/pic2mpg vdr-1.5.16/PLUGINS/src/pictures/pic2mpg
--- vdr-1.5.16~/PLUGINS/src/pictures/pic2mpg	2008-02-02 12:34:43.000000000 +0100
+++ vdr-1.5.16/PLUGINS/src/pictures/pic2mpg	2008-02-28 01:40:52.000000000 +0100
@@ -14,6 +14,7 @@
 use File::Path;
 use Getopt::Std;
 use Image::Size;
+use File::Temp qw/ tempfile tempdir /;
 
 $Usage = qq{
 Usage: $0 [options] picture-dir mpeg-dir
@@ -164,9 +165,23 @@
   my $Cmd = "$PNMCONV{$Type} $Pict 2> /dev/null |"
           . "pnmscale $verbose1 --xscale=$ScaleW --yscale=$ScaleH |"
           . "pnmpad $verbose1 --black --width $SW --height $SH |"
-          . "ppmntsc $verbose1 $system1 |"
-          . "ppmtoy4m $verbose2 -F $framerate -I p -S 420mpeg2 |"
+          . "ppmntsc $verbose1 $system1 |";
+
+  if (-e "/usr/bin/mpeg2enc") {
+    $Cmd .= "ppmtoy4m $verbose2 -F $framerate -I p -S 420mpeg2 |"
           . "mpeg2enc $verbose2 -f 3 -b 12500 -a $aspect -q 1 -n $system2 -o $Mpeg";
+  }
+  else
+  {
+    my $tempdir = tempdir("/tmp/pic2mpg-XXXX");
+    $Cmd .= "ppmtojpeg >$tempdir/01.jpg;"
+          . "ln -sf $tempdir/01.jpg $tempdir/02.jpg;"
+          . "ln -sf $tempdir/01.jpg $tempdir/03.jpg;"
+          . "ln -sf $tempdir/01.jpg $tempdir/04.jpg;"
+          . "ffmpeg -i $tempdir/%2d.jpg -f mpeg1video -y -b 12500 -r $framerate $Mpeg;"
+          . "rm -rf $tempdir";
+  }
+
   !system($Cmd) || die "$Cmd: $!\n";
   $Cmd = "touch -r $Pict $Mpeg";
   !system($Cmd) || die "$Cmd: $!\n";
